### PR TITLE
Add test cases for RDL (3 branches) in TextProtocolBackendHandlerFactoryTest

### LINKED ISSUE #7522: Enhance test cases for TextProtocolBackendHandlerFactory
The class TextProtocolBackendHandlerFactory has already split into multiple Factory classes. It is better to split test cases into suitable classes.

- [ ] Move TCL related test cases into `TransactionBackendHandlerFactoryTest`
- [ ] Keep one TCL SQL test case in `TextProtocolBackendHandlerFactoryTest` and assert return TransactionBackendHandler
- [ ] Move DAL related test cases into `DALBackendHandlerFactoryTest`
- [ ] Keep one DAL SQL test case in `TextProtocolBackendHandlerFactoryTest` and assert return UnicastBackendHandler
- [x] Add test cases for RDL (3 branches) in `TextProtocolBackendHandlerFactoryTest`
- [x] Add test case for MySQLShowTablesStatement in `DALBackendHandlerFactoryTest`
- [x] Add test case for CommitStatement in `TransactionBackendHandlerFactoryTest`
- [x] Add test case for RollbackStatement in `TransactionBackendHandlerFactoryTest`
- [x] Add test case for return BroadcastBackendHandler in `TransactionBackendHandlerFactoryTest`

### PR DESCRIPTION
**Pull request by tianhao960**
For #7522.

Changes proposed in this pull request:
- Add test cases for RDL (3 branches) in TextProtocolBackendHandlerFactory

### CODE REVIEW

**Review by terrymanu — CHANGES_REQUESTED** [review]
This issue is pretty old, some classes are changed.
Can you change the test class name to `RDLBackendHandlerFactory`?

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/handler/TextProtocolBackendHandlerFactoryTest.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.junit.Assert.assertThat;
+
+@RunWith(MockitoJUnitRunner.class)
+public class TextProtocolBackendHandlerFactoryTest extends ProxyContextRestorer {
```
Comment (terrymanu) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/handler/TextProtocolBackendHandlerFactoryTest.java:39]: Is it necessary to extends `ProxyContextRestorer`?

Comment (terrymanu) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/handler/TextProtocolBackendHandlerFactoryTest.java:39]: Please add final if class does not design for extend.

**Review by terrymanu — CHANGES_REQUESTED** [review]

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/handler/TextProtocolBackendHandlerFactoryTest.java (hunk 1 above)
Comment (tianhao960) [reply shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/handler/TextProtocolBackendHandlerFactoryTest.java:39]: ok, get it . I will move the test class to package org.apache.shardingsphere.proxy.backend.handler.distsql.rdl and change class name.

**Review by terrymanu — APPROVED** [review]

**Pull request merged**